config/main.py

```diff
@@ -702,6 +702,32 @@ def _restart_services():
     click.echo("Reloading Monit configuration ...")
     clicommon.run_command("sudo monit reload")
 
+def _get_delay_timers():
+    out = clicommon.run_command("systemctl list-dependencies sonic-delayed.target --plain |sed '1d'", return_cmd=True)
+    return [timer.strip() for timer in out.splitlines()]
+
+def _delay_timers_elapsed():
+    for timer in _get_delay_timers():
+        out = clicommon.run_command("systemctl show {} --property=LastTriggerUSecMonotonic --value".format(timer), return_cmd=True)
+        if out.strip() == "0":
+            return False
+    return True
+
+def _swss_ready():
+    out = clicommon.run_command("systemctl show swss.service --property ActiveState --value", return_cmd=True)
+    if out.strip() != "active":
+        return False
+    out = clicommon.run_command("systemctl show swss.service --property ActiveEnterTimestampMonotonic --value", return_cmd=True)
+    swss_up_time = float(out.strip())/1000000
+    now =  time.monotonic()
+    if (now - swss_up_time > 120):
+        return True
+    else:
+        return False
+
+def _system_running():
+    out = clicommon.run_command("sudo systemctl is-system-running", return_cmd=True)
+    return out.strip() == "running"
```

Comment:
While verifying PR https://github.com/Azure/sonic-buildimage/pull/8003. I hit this condition where I was I was not able to do "config reload". Even after system is up for a while the systemctl was showing running state as degraded
```
admin@str-s6000-acs-9:~$ sudo systemctl is-system-running
degraded
admin@str-s6000-acs-9:~$ sudo config reload -y
System is not up. Retry later or use -f to avoid system checks
admin@str-s6000-acs-9:~$ docker ps
CONTAINER ID        IMAGE                                COMMAND                  CREATED             STATUS              PORTS               NAMES
8baf489477e6        docker-snmp:latest                   "/usr/local/bin/supe…"   18 minutes ago      Up 2 minutes                            snmp
ad835c8b42ba        docker-sonic-mgmt-framework:latest   "/usr/local/bin/supe…"   18 minutes ago      Up 2 minutes                            mgmt-framework
930f188f20a7        docker-router-advertiser:latest      "/usr/bin/docker-ini…"   20 minutes ago      Up 5 minutes                            radv
15fd0337e6ea        docker-dhcp-relay:latest             "/usr/bin/docker_ini…"   20 minutes ago      Up 5 minutes                            dhcp_relay
c37a6752f57f        docker-lldp:latest                   "/usr/bin/docker-lld…"   20 minutes ago      Up 5 minutes                            lldp
79cb8660e7d2        docker-syncd-brcm:latest             "/usr/local/bin/supe…"   20 minutes ago      Up 5 minutes                            syncd
77e8ca15b220        docker-teamd:latest                  "/usr/local/bin/supe…"   20 minutes ago      Up 5 minutes                            teamd
ce63284a0f0a        docker-orchagent:latest              "/usr/bin/docker-ini…"   20 minutes ago      Up 5 minutes                            swss
277e43234314        docker-fpm-frr:latest                "/usr/bin/docker_ini…"   20 minutes ago      Up 5 minutes                            bgp
d9c63106202a        docker-platform-monitor:latest       "/usr/bin/docker_ini…"   21 minutes ago      Up 5 minutes                            pmon
34129d689ebd        docker-database:latest               "/usr/local/bin/dock…"   21 minutes ago      Up 6 minutes                            database
```
I have seen cases where system remains in degraded state as may be one of the systemd service is not started. @dgsudharsan do share your observations.

Reply:
@judyjoseph I believe we should try to fix the service that is in the degraded state. The whole point of introducing these checks were to avoid issues around config reload which happen when the system was degraded. Until the faulty services are fixed we can expect config reload without force not to work. However to give a handle we introduced '-f' option which lets user to take own risk and perform config reload when the system may not be in a stable state.